agents/agents-features/agents-features-snapshot/src/commonMain/kotlin/ai/koog/agents/snapshot/providers/PersistencyStorageProvider.kt

```diff
@@ -11,10 +11,10 @@ import ai.koog.agents.snapshot.feature.AgentCheckpointData
         "ai.koog.agents.snapshot.feature.PersistenceStorageProvider"
     )
 )
-public typealias PersistencyStorageProvider = PersistenceStorageProvider
+public typealias PersistencyStorageProvider<Filter> = PersistenceStorageProvider<Filter>
 
-public interface PersistenceStorageProvider {
-    public suspend fun getCheckpoints(agentId: String): List<AgentCheckpointData>
+public interface PersistenceStorageProvider<Filter> {
```

Comment:
Sorry, I fail to understand how this generic `Filter` can be used when you only know the most basic type, this `PersistenceStorageProvider` interface, and don't know the concrete implementation. If I only have an instance of this type, how can I pass a filter here, if It's essentially `Any?`? I think we need to redesign this part, to provide some generic `Filter` interface that wouldn't be tied to a particular in-memory filter operation, so it could be used in-memory, with a database, etc.